app/jobs/audit_manager.rb

```diff
@@ -0,0 +1,15 @@
+class AuditManager
```

Comment:
When I suggested making a standalone class, I may have been unnecessarily idealistic. These methods can go directly on `AuditFixityCheckJob` if we choose. There is a trade-off, here...

The ActiveJob subclasses behave somewhere between classes and instances in a way that is not obvious. When you call the `perform_later` method on the class, it enqueues a job (serializing all of the parameters, because the workers are in a separate Ruby process). When that job is processed, an instance of your job class is created with no parameters, and the `perform` method is called with the deserialized versions of the parameters you passed to `perform_later`. (phew!)

You can add other instance or class methods to the ActiveJob subclasses. I initially suggested a separate class to avoid any confusion about the perform behavior. However, having a class in the `jobs` directory that doesn't behave like the others is its own source of confusion. We could put this class somewhere else, but here, we're taking advantage of the conventional Rails directory structure and auto-loading.

Reconsidering, I think we should make this something more like:

```
class RepositoryAuditJob < ApplicationJob
  def perform(user: User.system_user, audit: Audit.new(user: user))
    # ...
  end
end
```

The invocation from the controller would be similar to what it is now:

```
  RepositoryAuditJob.perform_later(user: current_user, audit: audit)
```

And the "runner" invocation would take advantage of the default parameters:

```
bin/rails runner RepositoryAuditJob.perform_now
```

I will note, this is still a little awkward in the design because we expect a user, and an audit instance that knows the user... but it follows job conventions more closely, and is amenable to refactoring. The two-stage policy check in the controller is the issue there, and should be cleaned up such that the authorization does not depend on on Audit instance. Then, the Audit can be created wholly within in this job. That can happen now, or in another step.